paddle/platform/profiler.cc

```diff
@@ -169,5 +172,149 @@ std::vector<std::vector<Event>> DisableProfiler() {
   return result;
 }
 
+void PushEvent(const std::string& name, DeviceContext* dev_ctx) {
+  GetEventList().Record(EventKind::kPushRange, std::move(name), g_thread_id,
+                        dev_ctx);
+}
+
+void PopEvent(const std::string& name, DeviceContext* dev_ctx) {
+  GetEventList().Record(EventKind::kPopRange, std::move(name), g_thread_id,
+                        dev_ctx);
+}
+
+void ParseEvents(std::vector<std::vector<Event>>& events,
+                 EventSortingKey sorted_by) {
+  // Output header information
+  std::cout << "------------------------->"
+            << "     Profiling Report     "
+            << "<-------------------------"
+            << "\n\n";
+#ifdef PADDLE_WITH_CUDA
+  std::cout << "Place: GPU" << std::endl;
+#else
+  std::cout << "Place: CPU" << std::endl;
+#endif
```

Comment:
If compiling with GPU, the users also can run tasks on CPU only. So here should not use the `PADDLE_WITH_CUDA` macro.  You can use  `g_state` to determine.

Reply:
Done. Use another global variable `g_profiler_place` because `g_state` has been set to `kDisabled` when parsing events.